python-spec/src/somacore/scene.py

```diff
@@ -0,0 +1,270 @@
+    #     # of type annotations; it is (currently) expected.
+    #     class Scene(  # type: ignore[type-var]
+    #         ImplBaseCollection[ImplSOMAObject],
+    #         somacore.Scene[
+    #             ImplMultiscaleImage,
+    #             ImplPointCloud,
+    #             ImplGeometryDataFrame,
+    #             ImplSOMAObject,
+    #         ],
+    #     ):
+    #         ...
+
+    __slots__ = ()
+    soma_type: Final = "SOMAScene"  # type: ignore[misc]
+
+    img = _mixin.item[collection.Collection[_MultiscaleImage]]()
+    """A collection of multiscale images backing the spatial data.
+
+    Lifecycle: experimental
+    """
+
+    obsl = _mixin.item[collection.Collection[Union[_PointCloud, _GeometryDataFrame]]]()
+    """A collection of observation location data.
+
+    This collection exists to store any spatial data in the scene that joins on the obs
+    ``soma_joinid``. Each dataframe in ``obsl`` can be either a PointCloud
+    or a GeometryDataFrame.
+    
+    Lifecycle: experimental
+    """
+
+    varl = _mixin.item[
+        collection.Collection[
+            collection.Collection[Union[_PointCloud, _GeometryDataFrame]]
+        ]
+    ]()
+    """A collection of collections of variable location data.
+
+    This collection exists to store any spatial data in the scene that joins on the
+    variable ``soma_joinid`` for the measurements in the SOMA experiment. The top-level
+    collection maps from measurement name to a collection of dataframes.
+
+    Each dataframe in a ``varl`` subcollection can be either a GeometryDataFrame or a
+    PointCloud.
+    
+    Lifecycle: experimental
+    """
+
+    @property
+    @abc.abstractmethod
+    def coordinate_space(self) -> Optional[coordinates.CoordinateSpace]:
+        """Coordinate system for this scene.
+
+        Lifecycle: experimental
+        """
+        raise NotImplementedError()
+
+    @coordinate_space.setter
+    @abc.abstractmethod
+    def coordinate_space(self, value: coordinates.CoordinateSpace) -> None:
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def register_geometry_dataframe(
+        self,
+        key: str,
+        transform: coordinates.CoordinateTransform,
+        *,
+        subcollection: Union[str, Sequence[str]] = "obsl",
+        coordinate_space: Optional[coordinates.CoordinateSpace] = None,
+    ) -> _GeometryDataFrame:
+        """Adds the coordinate transform for the scene coordinate space to
+        a point cloud stored in the scene.
+
+        If the subcollection the geometry dataframe is inside of is more than one
+        layer deep, the input should be provided as a sequence of names. For example,
+        to register a geometry dataframe named  "transcripts" in the "var/RNA"
+        collection::
+
+            scene.register_geometry_dataframe(
+                'transcripts', transform, subcollection=['var', 'RNA'],
+            )
```

Comment:
Is this a workflow that is being supported? I thought the spec was pretty opinionated about where each kind of thing would be stored.

Reply:
The example in the docstring follows the storage guideline and is more than one layer deep.